worker/codalabworker/fsm.py

```diff
@@ -0,0 +1,96 @@
+import os
+import tempfile
+import shutil
+import pyjson
+
+class BaseStateCommitter(object):
+    def load(self):
+        """ Load and return the state """
+        raise NotImplementedError
+
+    def commit(self, state):
+        """ Commit the state """
+        raise NotImplementedError
+
+class JsonStateCommitter(BaseStateCommitter):
```

Comment:
what if the state file format needs to change? how will you handle upgrades aka schema migrations

Reply:
I'm not sure if handling that is worth the design/dev time. These state files are only for the recovery of the running jobs in case the worker dies mid job. In a controlled update setting we can either manually update/deal with the schema change at the update that changes it.

Reply:
okay